.github/workflows/upload.yml

```
@@ -29,7 +29,7 @@ jobs:
       ARCHIVE_NAME: "${{ github.event.repository.name }}-${{ github.ref_name }}"
     steps:
       - name: Checkout Source
-        uses: actions/checkout@v4
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633
```

https://github.com/actions/checkout/releases/tag/v4

```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
```